src/Microsoft.AspNetCore.Hosting/GenericHost/GenericWebHost.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    internal class GenericWebHost : IWebHost
+    {
+        private readonly IHost _host;
+
+        public GenericWebHost(IHost host)
+        {
+            _host = host;
+        }
+
+        public IFeatureCollection ServerFeatures => Services.GetRequiredService<IServer>().Features;
+
+        public IServiceProvider Services => _host.Services;
+
+        public void Dispose() => _host.Dispose();
+
+        public void Start() => _host.Start();
```

Comment:
Is this really right? Is the fact that this class is an implementation of `IWebHost` even significant since users can't call `.Build()`?

Reply:
I suppose this is here so you can resolve it as a service, but if you do - is it right to have these methods do what they do?

Reply:
Why wouldn't webhost just call base on build?